Meridian59/Data/Models/AdminData/AdminData.cs

```diff
@@ -0,0 +1,153 @@
+        public event AddTrackedAdminObjectEventHandler AdminObjectAdded;
+
+        protected void OnTrackedAdminObjectAdded(AddTrackedAdminObjectEventHandlerArgs e)
+        {
+            if (AdminObjectAdded != null) AdminObjectAdded(this, e);
+        }
+
+        //Event to pass a message back to the console
+        public delegate void LogAdminMessageEventHandler(LogAdminMessageEventHandlerArgs e);
+        public event LogAdminMessageEventHandler LogAdminMessage;
+        protected void OnLogAdminMessage(LogAdminMessageEventHandlerArgs e)
+        {
+            if (LogAdminMessage != null)
+            {
+                LogAdminMessage(e);
+            }
+        }
+        #endregion
+
+        void adminObjectProperty_ValueChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (sender.GetType() == typeof (AdminObjectProperty))
+            {
+                AdminObjectProperty changedProperty = (AdminObjectProperty)sender;
+                AdminObject changedAdminObject = changedProperty.GetOwner();
+
+                if (PacketSend != null)
+                {
+                    PacketSend(this,new GameMessageEventArgs(new ReqAdminMessage(string.Format("set object {0} {1} {2} {3}",
+                        changedAdminObject.ObjectNumber, changedProperty.PropertyName, changedProperty.PropertyType, changedProperty.PropertyValue))));
+                    PacketSend(this, new GameMessageEventArgs(new ReqAdminMessage(String.Format("show object {0}", changedAdminObject.ObjectNumber))));
+                }
+
+            }
+            else
+            {
+                throw new Exception("What?");
+            }
+        }
+
+        #region INotifyPropertyChanged
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        protected void RaisePropertyChanged(PropertyChangedEventArgs e)
+        {
+            if (PropertyChanged != null) PropertyChanged(this, e);
+        }
+
+        #endregion
+
+        #region IClearable
+        public void Clear(bool RaiseChangedEvent)
+        {
+            TrackedObjects.Clear();
+        }
+        #endregion
+
+        public void HandleAdminMessage(AdminMessage message)
+        {
+            if (message.Message.StartsWith(":< OBJECT"))
+            {
+                HandleAdminShowObjectMessage(message);
+            }
+            OnLogAdminMessage(new LogAdminMessageEventHandlerArgs(message.Message));
+        }
+        public void HandleAdminShowObjectMessage(AdminMessage message)
+        {
+            AdminObject obj = null;
+            bool track = false;
+
+            //Look for object info
+            var regex = new Regex(@":< OBJECT (?<objectnumber>\d*) is CLASS (?<classname>.*)");
+            var matches = regex.Matches(message.Message);
+            if (matches.Count != 1)
+            {
+                throw new Exception("Improper Response in HandleAdminShowObjectMessage()");
```

Comment:
Absolutely. I won't throw any in a finished product. Useful for debugging and troubleshooting.